modules/dlm/src/main/java/org/elasticsearch/dlm/action/PutDataLifecycleAction.java

```diff
@@ -37,7 +37,7 @@
 public class PutDataLifecycleAction extends ActionType<AcknowledgedResponse> {
 
     public static final PutDataLifecycleAction INSTANCE = new PutDataLifecycleAction();
-    public static final String NAME = "indices:admin/dlm/put";
+    public static final String NAME = "indices:admin/data_stream/lifecycle/put";
```

Comment:
As discussed on Slack, this should be fine for prod code since data lifecycles still sit behind a feature flag. Just need to confirm that this doesn't introduce unexpected test failures in BWC-test suites. I don't see any issues with this change from a security perspective.